src/pages/groupComparison/FisherExactTwoSidedTestLabel.tsx

```diff
@@ -0,0 +1,100 @@
+import { action, makeObservable, observable } from 'mobx';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import MutationMapperDataStore from 'shared/components/mutationMapper/MutationMapperDataStore';
+import { ComparisonGroup } from './GroupComparisonUtils';
+import { toConditionalPrecisionWithMinimum } from 'shared/lib/FormatUtils';
+import _ from 'lodash';
+import { getTwoTailedPValue } from 'shared/lib/FisherExactTestCalculator';
+import { formatPercentValue } from 'cbioportal-utils';
+import intersect from 'fast_array_intersect';
+import InfoIcon from 'shared/components/InfoIcon';
+import { MutationOverlapOverlay } from 'shared/components/mutationTable/column/mutationOverlap/MutationOverlapOverlay';
+
+interface IFisherExactTwoSidedTestLabelProps {
+    dataStore: MutationMapperDataStore;
+    hugoGeneSymbol: string;
+    groups: ComparisonGroup[];
+    profiledPatientCounts: number[];
+}
+
+export const FisherExactTwoSidedTestLabel: React.FC<IFisherExactTwoSidedTestLabelProps> = observer(
+    ({
+        dataStore,
+        hugoGeneSymbol,
+        groups,
+        profiledPatientCounts,
+    }: IFisherExactTwoSidedTestLabelProps) => {
+        const groupAMutatedCount = _.uniq(
+            _.flatten(
```

Comment:
simplify this code (try resist nesting when possible). and put a comment what operation we're doing.